File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
    let prevarr = arr.slice();
    var answer = 0;
    while(true){
        arr = arr.map(v => {
            if(v >= 50 && v % 2 === 0) {
                v /= 2;
                return v;
            }
            else if(v < 50 && v % 2 === 1) {
                v = v * 2 + 1;
                return v;
            }
        })
        
        if (JSON.stringify(prevarr) === JSON.stringify(arr)) break;
        prevarr = arr.slice();
        answer += 1;
    }
    answer--;
    return answer;
}

File: 프로그래머스/unrated/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> op(vector<int> arr){
    for(int i = 0; i<arr.size(); i++){
        if(arr[i] >= 50 && arr[i] % 2 == 0) 
            arr[i] /= 2;
        else if(arr[i] < 50 && arr[i] % 2 == 1)
            arr[i] = arr[i] * 2 + 1;
    }
    return arr;
}

int solution(vector<int> arr) {
    vector<int> prevarr = arr;
    int answer = 0;
    while(true){
        arr = op(arr);
        if(arr == prevarr)
            break;
        else {
            answer++;
            prevarr = arr;
        }
    }

    return answer;
}